src/ja/piccoma/src/eu/kanade/tachiyomi/extension/ja/piccoma/Piccoma.kt

```diff
@@ -0,0 +1,245 @@
+package eu.kanade.tachiyomi.extension.ja.piccoma
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.asObservable
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import keiyoushi.utils.firstInstance
+import keiyoushi.utils.parseAs
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import rx.Observable
+
+class Piccoma : HttpSource() {
+    override val name = "Piccoma"
+    override val baseUrl = "https://piccoma.com"
+    override val lang = "ja"
+    override val supportsLatest = true
+
+    override val client = network.cloudflareClient.newBuilder()
+        .addInterceptor(ImageInterceptor())
+        .build()
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/web/ranking/K/P/0", headers)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val mangas = document.select("section.PCM-productRanking li > a").map { element ->
+            SManga.create().apply {
+                url = element.attr("href")
```

Comment:
```suggestion
                setUrlWithoutDomain(element.absUrl("href"))
```

Yes, you should use both `absUrl()` and strip it with `setUrlWithoutDomain()`. Jsoup knows the context to resolve relative paths correctly, and you don't want to [reimplement removing the domain](https://github.com/mihonapp/mihon/blob/e8c6e3364dded0cb21df572be175cbb01adc243f/source-api/src/commonMain/kotlin/eu/kanade/tachiyomi/source/online/HttpSource.kt#L414-L428).